cmd/thanos/query.go

```diff
@@ -345,6 +370,8 @@ func runQuery(
 	enableTargetPartialResponse bool,
 	enableMetricMetadataPartialResponse bool,
 	enableExemplarPartialResponse bool,
+	fileSDConfig httpconfig.FileSDConfig,
```

Comment:
You may not need to pass this, what I suggest is to directly pass the global file sd options, i.e`*fileSDFiles` and `*fileSDInterval` as arguments, so that you can combine that in `query.LoadConfig`.

Reply:
Done!